test.py

```diff
@@ -1,41 +1,174 @@
+        skipIf,
+        condition=not noqa_supported,
+        reason='noqa is not supported on this flake8 version')
+else:
+    # Python 2.6 does not have skipIf or SkipTest, so
+    # completely skip the test which will be reported as success.
+    def skip_if_noqa_unsupported():
+        """Decorator to unconditionally skip test method."""
+        def noop(*args, **kwargs):
+            pass
+
+        return noop
+
+
+T001s = 'T001 print statement found.'
+T001f = 'T001 print function found.'
+T101 = 'T101 Python 2.x reserved word print used.'
+
+
+class TestNoQA(Flake8PrintTestCases):
+
+    @skip_if_noqa_unsupported()
     def test_skips_noqa(self):
         result = check_code_for_print_statements('print(4) # noqa')
         assert_equal(result, list())
 
+    @skip_if_noqa_unsupported()
+    def test_skips_noqa_multiline_end(self):
+        result = check_code_for_print_statements(dedent("""
+            print("a"
+                  "b")  # noqa
+        """))
+        assert_equal(result, list())
+
+    @skip_if_noqa_unsupported()
+    def test_skips_noqa_multiline_middle(self):
+        result = check_code_for_print_statements(dedent("""
+            print("a"
+                  "b"  # noqa
+                  "c")
+        """))
+        assert_equal(result, list())
+
+    @skip_if_noqa_unsupported()
+    def test_skips_noqa_multiline_start(self):
+        result = check_code_for_print_statements(dedent("""
+            print("a"  # noqa
+                  "b")
+        """))
+        assert_equal(result, list())
+
+    @skip_if_noqa_unsupported()
     def test_skips_noqa_line_only(self):
-        result = check_code_for_print_statements('print(4) # noqa\nprint(5)\n # noqa')
-        try:
-            assert_equal(result, [{'col': 0, 'line': 2, 'message': 'T001 print statement found.'}])
-        except AssertionError:
-            assert_equal(result, [{'col': 5, 'line': 2, 'message': 'T001 print statement found.'}])
+        result = check_code_for_print_statements('print(4); # noqa\nprint(5)\n # noqa')
+        assert_equal(result, [{'col': 0, 'line': 2, 'message': T001f}])
+
+
+class TestGenericCases(Flake8PrintTestCases):
+
+    def test_catches_multiline_print(self):
+        result = check_code_for_print_statements(dedent("""
+            print("a"
+                  "b")
+        """))
+        assert_equal(result, [{'col': 0, 'line': 2, 'message': T001f}])
 
     def test_catches_simple_print_python2(self):
-        try:
-            result = check_code_for_print_statements('print 4')
-            assert_equal(result, [{'col': 0, 'line': 1, 'message': 'T001 print statement found.'}])
-        except SyntaxError:
-            from sys import version
-            assert_true(float(version[:3]) >= 3.0)
+        result = check_code_for_print_statements('print 4')
+        assert_equal(result, [{'col': 0, 'line': 1, 'message': T001s}])
+
+    def test_catches_empty_print_python2(self):
```

Comment:
some new test methods here, for empty prints; this one is captured as T101 instead of T001, which will be a difference from the `ast` version of flake8-print